add support for 2CH-Triac-HAT

DanielDecker opened the pull request:
This PR adds support and documentation for the [2Ch-Triac-HAT](https://www.waveshare.com/wiki/2-CH_TRIAC_HAT) from Waveshare via i2c-bus. The Triac supports continuous dimming on both channels.
The Triac actuator was tested with all connectors.

This will also update the readme's for all connectors so that all tables are easily readable in the source code.

rkoshak reviewed (commented):
Looks good to me. Just a couple of minor nit picks and one bit of code to look closely at to make sure there isn't a problem with a never ending while loop possible.

DanielDecker commented on i2c/triac.py:
```diff
@@ -0,0 +1,407 @@
+            #ignore the DIM command if smooth_change_in_progress
+            if not self.dimmer.smooth_change_in_progress:
+                if isinstance(self.dimmer.thread, Thread):
+                    #Info: waiting for the thread will rais toggle time for local connections,
```
I'll fix this.

DanielDecker commented on i2c/triac.py:
```diff
@@ -0,0 +1,407 @@
+    """
+    def __init__(self):
+        self.bus = smbus2.SMBus(1)   # 0 = /dev/i2c-0 (port I2C0), 1 = /dev/i2c-1 (port I2C1)
+        # the triac uses bus address 0x47, the HAT is hardcoded to this address
```
And this too.

DanielDecker commented on i2c/triac.py:
```diff
@@ -0,0 +1,407 @@
+            return
+        elif msg == "STOP":
+            #make sure we don't interrupt the smooth change
+            if not self.dimmer.smooth_change_in_progress:
```
`_smoot_dimmer()` is called in two cases:

1. The actuator receives ON / OFF or a nummer e.g. 55
2. The actuator receives DIM

In the first case, the actuator will smoothly change the PWM to 55%. To avoid interruption by DIM and STOP commands, `self.dimmer.smooth_change_in_progress` is checked beforehand. This allows the actuator to be connected to a RpiGPIOSensor. `self.dimmer.smooth_change_in_progress` is initialized to False in `__init__()`.

If `self.dimmer.stop_thread` is never set to True, `_smoot_dimmer()` will run until the target_value is reached e. g. 100 and exit. `self.dimmer.stop_thread` is only needed to exit early.

>All of these flags setting are spread throughout the code which makes me nervous. Is there a way to create transactions somehow or at least move the setting to true and false of theses flags closer together? 

I'm not sure how to archive this. Should I refractor the if statement below to make the structure more clearer?
```python
if msg.isdigit():
   #code
elif msg == "ON":
   #code
elif msg == "OFF":
   #code
elif msg == "DIM":
   #code
elif msg == "STOP":
   #code
elif is_toggle_cmd(msg):
   #code
else:
   #code
```

Thank you for reviewing the code! =)

DanielDecker commented:
- I moved the thread control from the TriacDimmer class to a new one and moved the de-bounce code to utils to clean up TriacDimmer
- The RpiGpioActuator also uses the new de-bounce class

:x: The RelayHat is not jet migrated to the de-bounce class, as I currently don't have the hardware with me at the moment, so I can't test the changes.

The above commits are tested with all connectors.
I hope the Triac code is now easier to understand! :-)

DanielDecker commented:
I'm glad to read that. :-)
Please merge this PR now. I'll create a new one for the RelayHat changes.
